### PR TITLE
chore(node): use parking lot and setup deadlock detection

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

**Comment by joshuef** [comment]
bors try

**Comment by joshuef** [comment]
bors try

File: sn_node/src/bin/sn_node/main.rs [hunk 1]
```diff
@@ -61,6 +61,34 @@ fn main() -> Result<()> {
         (rt, guard)
     };
 
+    {
+        // only for #[cfg]
```
Comment (RolandSherwin) [inline sn_node/src/bin/sn_node/main.rs:65]: Is the deadlock detection thread meant to be enabled as an optional feature? If so, I think the flag is absent here...

Comment (joshuef) [reply sn_node/src/bin/sn_node/main.rs:65]: At the moment, I think we just enable it outright. We can disable w/ feat flag later when we're confident we're not seeing it.

**Review by oetyng — APPROVED** [review]
LGTM. Just the doc / sleep time diff, left a comment there.

File: sn_node/src/bin/sn_node/main.rs [hunk 2]
```diff
@@ -61,6 +61,33 @@ fn main() -> Result<()> {
         (rt, guard)
     };
 
+    {
+        use parking_lot::deadlock;
+        use std::thread;
+
+        // Create a background thread which checks for deadlocks every 10s
+        let _handle = thread::spawn(move || loop {
+            thread::sleep(Duration::from_secs(3));
```
[the remark this answers is not in the document]
Comment (joshuef) [reply sn_node/src/bin/sn_node/main.rs:70]: ah, 10 was the copypasta. 3 was thinking, it'd be nice to know more _sooner_. (I think this is how often we report of a stall... if it cleans itself up before, then nothing is reported?)

i'll correct the msg 👍 thanks

**Comment by joshuef** [comment]
bors try

**Comment by joshuef** [comment]
bors try

**Pull request merged**